docs/samples/browser-plugin-meetings/app.js

```diff
@@ -1255,6 +1328,434 @@ function clearMediaDeviceList() {
+  const meeting = getCurrentMeeting();
+
+  document.getElementById('stage-selector-wrapper').classList.remove('hidden');
+
+  const selectEl = document.getElementById('multistream-stage-selector');
+
+  // clear out all the options
+  while (selectEl.firstChild) {
+    selectEl.removeChild(selectEl.lastChild);
+  }
+
+  // create new options - 1 for each member
+  const members = meeting.members.membersCollection.getAll();
+
+  Object.entries(members).forEach(([key, member]) => {
+    if (member.status !== 'NOT_IN_MEETING') {
+      addElement(selectEl, 'option', {value: member.id, innerText: member.name});
+    }
+  });
+
+  selectEl.selectedIndex = 0;
+}
+
+function hideStageSelector() {
+  document.getElementById('stage-selector-wrapper').classList.add('hidden');
+}
+
+function updateMultistreamVideoLayout() {
+  const meeting = getCurrentMeeting();
+
+  if (!meeting) {
+    return;
+  }
+  if (!meeting.mediaProperties.webrtcMediaConnection) {
+    return;
+  }
+
+  const oldLayoutId = meeting.remoteMediaManager?.getLayoutId();
+  const newLayoutId = multistreamLayoutElm.value;
+
+  if (newLayoutId === oldLayoutId) {
+    return;
+  }
+
+  console.log(`MeetingStreams#updateMultistreamVideoLayout() :: changing layout from ${oldLayoutId} to ${newLayoutId}`);
+
+  if (newLayoutId === 'Stage') {
+    showStageSelector();
+  }
+  else if (oldLayoutId === 'Stage') {
+    hideStageSelector();
+
+    // we also undo all of the staging of members when we switch to a non-staged layout
+    resetMultistreamStage(meeting);
+  }
+
+  if (meeting.remoteMediaManager) {
+    meeting.remoteMediaManager.setLayout(newLayoutId);
+  }
+}
+
+function processNewVideoPane(meeting, paneGroupId, paneId, remoteMedia) {
+  const videoPane = allVideoPanes[paneGroupId][paneId];
+
+  videoPane.remoteMedia = remoteMedia;
+
+  remoteMedia.on('sourceUpdate', (data) => {
+    updateVideoPane(videoPane, meeting, data.state, data.memberId, `${paneGroupId}.${paneId} ${remoteMedia.id}`, 'update');
+  });
+
+  remoteMedia.on('mediaStopped', (data) => {
+    updateVideoPane(videoPane, meeting, 'no source', undefined, `${paneGroupId}.${paneId} ${remoteMedia.id}`, 'mediaStopped');
+  });
+
+  remoteMedia.on('mediaStarted', (data) => {
+    videoPane.videoEl.srcObject = data.stream;
+    updateVideoPane(videoPane, meeting, remoteMedia.sourceState, remoteMedia.memberId, `${paneGroupId}.${paneId} ${remoteMedia.id}`, 'mediaStarted');
+  });
+}
+
+function setupMultistreamEventListeners(meeting) {
+  meeting.on('media:remoteAudio:created', (audioMediaGroup) => {
+    console.log('MeetingStreams#setupMultistreamEventListeners :: got AUDIO remote media group created');
+    audioMediaGroup.getRemoteMedia().forEach((media, index) => {
+      document.getElementsByClassName('multistream-remote-audio')[index].srcObject = media.stream;
+    });
+  });
+
+  meeting.on('media:remoteVideo:layoutChanged', ({layoutId, activeSpeakerVideoPanes, memberVideoPanes}) => {
```

Comment:
Shouldn't this be on the layoutManger

Reply:
The issue is that the layout manager is only created when `addMedia()` is called (because its constructor needs the configuration that's passed in as `addMedia()` argument), so you can't register for events on it beforehand, because it doesn't exist yet, and when you register after `addMedia()` resolves, it is too late and you miss the initial layout notification.
However, if you think it's important to have this fixed, we can change it so that layout manager is always created in the meeting constructor, but it would be just empty, without a config (or with a default config) and then when `addMedia()` is called, it can update the config by passing it through a call to RemoteMediaManager.start()

Reply:
agreed offline that for now we keep it as it is and will re-evaluate this again later when defining API for local media